### PR TITLE
refactor: remove sub. notification and webhook connector GenServers

### PR DESCRIPTION
**Pull request by w3b6x9**
## What kind of change does this PR introduce?

refactor

## What is the current behavior?

`SubscribersNotification`, `Connectors`, and `WebhookConnector` are `GenServer`s and use up a lot of memory.

Related issue: #125

## What is the new behavior?

- `Connectors` has been removed
- `SubscribersNotification` and `WebhookConnector` are no longer `GenServer`s

Related comment: https://github.com/supabase/realtime/issues/125#issuecomment-781772587

## Additional context

### CODE REVIEW

File: server/lib/realtime/subscribers_notification.ex [hunk 1]
```diff
@@ -41,107 +36,112 @@ defmodule Realtime.SubscribersNotification do
...
-          do: "#{schema_keys}:#{table_keys}:#{column_keys}"
+          do: [schema_keys, ":", table_keys, ":", column_keys] |> IO.iodata_to_binary()
 
     Enum.any?(config, fn c -> c.relation in valid_patterns end)
   end
 
-  defp is_valid_notification_key(v) do
-    v != nil and v != :unchanged_toast and String.length(v) < 100
+  defp is_valid_notification_key(v) when is_binary(v) do
+    String.length(v) < 100
```
Comment (kiwicopple) [inline server/lib/realtime/subscribers_notification.ex:143]: I'm guessing the `is_binary` covers any cases where Postgres sends out an `:unchanged_toast` event?

File: server/test/realtime/webhook_connector_test.exs [hunk 2]
```diff
@@ -1,62 +1,149 @@
...
+          second: 19,
+          std_offset: 0,
+          time_zone: "Etc/UTC",
+          utc_offset: 0,
+          year: 2021,
+          zone_abbr: "UTC"
+        },
+        record: %{
+          "details" =>
+            "The SCSI system is down, program the haptic microchip so we can back up the SAS circuit!",
```
Comment (kiwicopple) [inline server/test/realtime/webhook_connector_test.exs:52]: :)

**Comment by kiwicopple** [comment]
Really nice work @w3b6x9 . very exciting change 🚀

File: server/lib/realtime/subscribers_notification.ex (hunk 1 above)
Comment (w3b6x9) [reply server/lib/realtime/subscribers_notification.ex:143]: @kiwicopple yep! I also made sure to have that covered in tests: https://github.com/supabase/realtime/blob/refactor/remove-genservers/server/test/realtime/subscribers_notification_test.exs#L530

File: server/test/realtime/webhook_connector_test.exs (hunk 2 above)
Comment (w3b6x9) [reply server/test/realtime/webhook_connector_test.exs:52]: @kiwicopple I wish I came up with it! All @fracek!

**Pull request merged**